### PR TITLE
Add Assets module

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: core/client/db/src/lib.rs [hunk 1]
```diff
@@ -791,7 +791,7 @@ impl<Block> client::backend::Backend<Block, Blake2Hasher> for Backend<Block> whe
 			match self.storage.state_db.revert_one() {
 				Some(commit) => {
 					apply_state_commit(&mut transaction, commit);
-					let removed = best.clone();
+					let _removed = best.clone();
```
Comment (bkchr) [inline core/client/db/src/lib.rs]: Maybe we just want to remove it completely, when it is not used anyway.

File: srml/assets/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,228 @@
+// Copyright 2017-2018 Parity Technologies (UK) Ltd.
```
Comment (bkchr) [inline srml/assets/src/lib.rs]: Nitpick, you started this probably 2018^^

File: core/client/db/src/lib.rs (hunk 1 above)
Comment (gavofyork) [reply core/client/db/src/lib.rs]: just wanted to silence the wanting - would keep full removal for a followup

File: srml/assets/src/lib.rs (hunk 2 above)
Comment (gavofyork) [reply srml/assets/src/lib.rs]: i think it's fine to keep copyright/license headers consistent throughout

File: srml/assets/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Assets {
+		/// The number of units of assets held by any given account.
+		Balances: map (AssetId, T::AccountId) => T::Balance;
+		/// The next asset identifier up for grabs.
+		NextAssetId get(next_asset_id): AssetId;
+
+		// Required for compailation @bkchr needs fixing! :)
+		Foo get(foo) config(): T::Balance;
```
Comment (gavofyork) [inline srml/assets/src/lib.rs]: @bkchr this needs addressing :)

Comment (bkchr) [reply srml/assets/src/lib.rs]: I know, I'm already on it and have working implementation. Just need some more tweaks. Will push it to this branch :)

**Comment by bkchr** [comment]
@gavofyork I pushed the new commit that contains the fix for the `GenesisConfig`.

**Pull request merged**